### PR TITLE
Fix position_ids docstring in modeling_flash_attention_utils.py

### LINKED ISSUE #44373: Wrong docstring for position_ids
### System Info

latest commit, see link below

### Who can help?

@stevhliu 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Wrong docstring:

https://github.com/huggingface/transformers/blob/11b1906d5c0dae39c13270e47cc02c4cde70e548/src/transformers/modeling_flash_attention_utils.py#L359-L360

https://github.com/huggingface/transformers/blob/11b1906d5c0dae39c13270e47cc02c4cde70e548/src/transformers/modeling_flash_attention_utils.py#L412-L413

The docstring is for attention_mask, not position_ids

https://github.com/huggingface/transformers/blob/11b1906d5c0dae39c13270e47cc02c4cde70e548/src/transformers/modeling_flash_attention_utils.py#L201

### Expected behavior

Expected docstring:

https://github.com/huggingface/transformers/blob/11b1906d5c0dae39c13270e47cc02c4cde70e548/src/transformers/models/layoutlmv2/modeling_layoutlmv2.py#L897-L901

or

https://github.com/huggingface/transformers/blob/11b1906d5c0dae39c13270e47cc02c4cde70e548/src/transformers/masking_utils.py#L715-L716

### PR DESCRIPTION
**Pull request by mvanhorn**
Fixes #44373

## Summary
- Corrected the docstring for `position_ids` parameter in `prepare_fa_kwargs_from_position_ids` and `_prepare_from_posids` which incorrectly described attention mask semantics ("Boolean or int tensor... 1 means valid and 0 means not valid")
- The docstring now accurately describes position indices behavior

## Testing
- Docstring-only change, no code behavior affected

This contribution was developed with AI assistance (Claude Code).

### CODE REVIEW

File: src/transformers/modeling_flash_attention_utils.py [hunk 1]
```diff
@@ -365,7 +365,8 @@ def prepare_fa_kwargs_from_position_ids(position_ids):
 
     Arguments:
         position_ids (`torch.Tensor`):
-            Boolean or int tensor of shape (batch_size, sequence_length), 1 means valid and 0 means not valid.
+            Indices of positions of each input sequence tokens in the position embeddings. Selected in the range
+            `[0, config.n_positions - 1]`. Shape: (batch_size, sequence_length).
```
Comment (vasqu) [inline src/transformers/modeling_flash_attention_utils.py:369]:
```suggestion
            Indices of positions of each input sequence tokens. Shape: (batch_size, sequence_length).
```
I'd rather follow https://github.com/huggingface/transformers/blob/a049c0044867607b697aaef55de6f87df2120baa/src/transformers/utils/generic.py#L756-L757

The generic docstring is a bit outdated imo, position embeddings sounds like the old absolute embeddings + the range can be indefinite since rope can be extended etc

Comment (vasqu) [reply src/transformers/modeling_flash_attention_utils.py:369]: Same below then

**Review by vasqu — COMMENTED** [review]
Thanks, let's update the strings a bit to a more simple one

File: src/transformers/modeling_flash_attention_utils.py [hunk 2]
```diff
@@ -365,7 +365,8 @@ def prepare_fa_kwargs_from_position_ids(position_ids):
...
```
Comment (stevhliu) [inline src/transformers/modeling_flash_attention_utils.py]:
```suggestion
            Tensor of shape `(batch_size, sequence_length)` containing position indices of each input sequence tokens in the position embeddings.
```

**Review by stevhliu — APPROVED** [review]
one minor comment, but lgtm!

**Comment by mvanhorn** [comment]
Thanks @vasqu and @stevhliu for the reviews\! Simplified both docstrings to drop the "position embeddings" reference and range constraint per @vasqu's suggestion - updated in both `prepare_fa_kwargs_from_position_ids` and `_prepare_from_posids`. Pushed in 64de4f4.

**Comment by mvanhorn** [comment]
@vasqu @stevhliu - friendly ping. Approved for a while with no merge activity - anything else needed on my end?

**Comment by qgallouedec** [comment]
This contribution comes from a profile that opens a massive number of pull requests generated purely by AI; I'm closing it.
